packages/block-library/src/video/edit-common-settings.js

```diff
@@ -21,13 +21,15 @@ const VideoSettings = ( { setAttributes, attributes } ) => {
 		preload,
 	} = attributes;
 
-	const getAutoplayHelp = useCallback( ( checked ) => {
-		return checked
-			? __(
-					'Note: Autoplaying videos may cause usability issues for some visitors.'
-			  )
-			: null;
-	}, [] );
+	const autoPlayHelpText = __(
+		'Note: Autoplaying videos may cause usability issues for some visitors.'
+	);
+	const getAutoplayHelp = Platform.select( {
+		web: useCallback( ( checked ) => {
+			return checked ? autoPlayHelpText : null;
+		}, [] ),
```

Comment:
@jd-alexander I've been reading about the `useCallback` hook recently and read that it can be a memory hog and should only be used in complex functions where caching would be beneficial for performance reasons. Since this is a really simple function I'm curious if why you're using it here. I'm still learning so please excuse the newbie question if it's obvious.

Reply:
Hi @AmandaRiu great observation. Since the `useCallback` behavior existed before I didn't remove it because it is providing performance improvements for the Video block's settings. If you look at this [PR](https://github.com/WordPress/gutenberg/pull/28220) that introduced the improvements you will notice that the updating time was reduced from 8.09ms to 2.74ms since we are avoiding redundant updates by caching the `checked` value.

Reply:
Oh fascinating. Thanks for the great info @jd-alexander! 👍

Reply:
No prob 🙇🏾

Reply:
Not sure if this is satisfying the ["Only Call Hooks at the Top Level"](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) rule 🤔

Reply:
> Not sure if this is satisfying the "Only Call Hooks at the Top Level" rule 🤔

is it because there's the usage of `useCallback` within the `Platform.select` method? Could you expound on how it is not satisfying the rule? Based on my understanding, I think that the behavior here is deterministic because `useCallback` will always be consistently triggered on the web only even though it is within conditional logic. Nonetheless, I may be missing something so let me know what you think about my thoughts here.

Reply:
> is it because there's the usage of useCallback within the Platform.select method? 
That's it 👍 I was a bit confused, as it's kind of a condition and the hook is not at the top level, but I think you're right that it shouldn't break the rule as it will be always be triggered on the web no matter 👍

Reply:
> That's it 👍 I was a bit confused, as it's kind of a condition and the hook is not at the top level, but I think you're right that it shouldn't break the rule as it will be always be triggered on the web no matter 👍

No problem 👍🏾